taskfile/read/taskfile.go

```diff
@@ -113,6 +113,11 @@ func Taskfile(dir string, entrypoint string) (*taskfile.Taskfile, error) {
 		if err = taskfile.Merge(t, includedTaskfile, namespace); err != nil {
 			return err
 		}
+
+		if t.Tasks[namespace] == nil && includedTaskfile.Tasks["default"] != nil {
+			t.Tasks[namespace] = includedTaskfile.Tasks["default"]
+		}
```

Comment:
I think this implementation has unintended side-effects. Tasks assigned here will be duplicated on `task --list`.

I added a `default` task to our `docs/Taskfile.yml` as an example. See the duplicated `docs:default`:

```shell
$ task --list
task: Available tasks for this project:
* clean: 		Cleans temp files and folders
* cli-deps: 		Downloads CLI dependencies
* docs:default: 	Serves the documentation site locally
* docs:default: 	Serves the documentation site locally
* docs:ico: 		Generate favicon.ico from Logo.png
* docs:install: 	Installs docsify to work the on the documentation site
* docs:serve: 		Serves the documentation site locally
* gen-install-script: 	Generate install script using https://github.com/goreleaser/godownloader
* install: 		Installs Task
* lint: 		Runs golint
* mod: 			Downloads and tidy Go modules
* test: 		Runs test suite
* test-release: 	Tests release process without publishing
```

My proposal is that we implement this near the code below, by looking for `foo:default` if a `foo` task is not available:

https://github.com/go-task/task/blob/2373743eacee1d6d38f8f97047d241204bbb5061/variables.go#L25-L28

@tylermmorton What do you think? Do you have a different opinion?